### ISSUE TITLE
Running multiple socket.io instances, Redis pub/sub and Nginx

### ISSUE THREAD

**Issue by http91** [socketio/socket.io#2539]
Hey.

I [read](http://socket.io/docs/using-multiple-nodes/) the docs.

TL;DR
- I want to use socket.io. I attach my socketio server to my Express.js app to do that.
- I need to run multiple instances on different ports. No problem here. I followed the docs for Nginx and everything works fine.
- I decided **not to use** Node's own cluster, I'll use Nginx's for load balancing (that's why I create multiple instances of the app and let Nginx handle load balancing). It supports websockets, so this one is also sorted out.
- There'll be multiple instances of the app, they can't talk to each other since they run on different ports, so I need to use Redis' pub/sub mechanism as a wrapper in order to mimick socketio's emit & broadcast functionality between them. That way even if I have multiple instances of an app or run it on different servers with multiple instances, everybody (by everybody I mean both people and instances) will be able to talk to each other as long as they're connected to the same Redis server. To achieve this, I'll need socket.io-redis and socket.io-emitter modules. _Sticky session module is required only if you're dealing with Node's own cluster. In my case, it's not required._

I'm going to give it a go, but I wanted to ask before trying it. Have I got that right?

**Comment by darrachequesne** [comment]
That issue was closed automatically. Please check if your issue is fixed with the latest release, and reopen if needed (with a [fiddle](https://github.com/darrachequesne/socket.io-fiddle) reproducing the issue if possible).

**Comment by ancashoria** [comment]
hey @http91 
did you manage to solve this problem?
I'm also trying to figure out how to have multiple node instances and still have proper client communication over websockets

**Comment by http91** [comment]
@ancashoria

It has been a very long time since I've tried to do it, I don't work with high performing, "I need everything there is" type of applications anymore, but this might help : 

https://stackoverflow.com/questions/36961882/socket-io-and-redis-pub-sub

**Comment by ancashoria** [comment]
Thanks @http91 
I'm in the same boat as you were back then. I could do fine with one node instance and websocket communication, but I want to have a scalable solution.

**Comment by darrachequesne** [comment]
Please take a look at https://github.com/socketio/socket.io/tree/master/examples/cluster-nginx.

https://github.com/darrachequesne/socket.io-proxy may help too.
